waku/common/enr.nim

```diff
@@ -0,0 +1,107 @@
+## An extension wrapper around nim-eth's ENR module
```

Comment:
Any reason why its on waku/common? Old code was in waku/v2.

Reply:
> Any reason why its on waku/common?

We are placing in the `common` module functionality that potentially:

- Could be shared among Waku v1 and v2.
- Could be shared among different submodules within `waku/v2` module.
- Could be upstreamed to the library that they are extending (e.g., this case is extending the ENR functionality of nim-eth module).

> Old code was in waku/v2.

There is no such thing as old code here. This is not a code refactoring PR 👀 This is a new addition to the code base that conveniently extends nim-eth functionality by wrapping the `Record.init()` proc aiming to make it externally extensible.